sql/ballots/ballots.go

```diff
@@ -74,72 +73,67 @@ func LoadBlob(ctx context.Context, db sql.Executor, id []byte, b *sql.Blob) erro
 }
 
 // Get ballot with id from database.
-func Get(db sql.Executor, id types.BallotID) (rst *types.Ballot, err error) {
-	if rows, err := db.Exec(`select ballot, length(identities.proof)
-	from ballots left join identities using(pubkey)
-	where id = ?1;`,
+func Get(tx sql.Transaction, id types.BallotID) (rst *types.Ballot, err error) {
+	var dErr error
+	rows, err := tx.Exec("select ballot from ballots where id = ?1",
```

Comment:
Why is a transaction needed here? I think it is not necessary to decode the ballot and check malfeasance atomically. It should be fine to load the ballot and then check for malfeasance without a TX IMO. 

Reasoning: an identity might become malicious at any moment. So it is possible that the ID becomes malicious right after the TX ends. Making it atomic doesn't give any additional benefits.

Reply:
Sure, I aimed at preserving the behavior from before my changes (i.e. check is a single query/transaction) but I can change this to not use a transaction.

Reply:
Dropped the requirement for a transaction.